_episodes/02-image-basics.md

```diff
@@ -97,54 +60,214 @@ color, but that each pixel can have a different color from its neighbors.
+
+Now that we have our libraries loaded, we will run a Jupyter Magic Command that will insure our images display in our juptyer document with pixel information that will help us more efficiently run commands later in the session.
+ ~~~
+%matplotlib widget
+~~~
+{: .language-python}
+
+With that taken care of, let's load our image data from disk using the imread function from the skimage.io library and display it using the imshow function from the matplotlib library.
+ ~~~
+image = skimage.io.imread(fname="data/eight.tif")
+plt.imshow(image)
+~~~
+{: .language-python}
+![Image of 8](../fig/02-eight.png)
+You might be thinking, "That does look vaguely like an eight, and I see two colors but how can that be only 15 pixels". The display of the eight you see does use a lot more screen pixels to display our eight so large, but that doesn not mean there is information for all those screen pixels  in the file. All those extra pixels are a consequence of our viewer creating additional pixels through interpolation. It could have just displayed it as a tiny image using only 15 screen pixels if the viewer was written differently.
+
+While many imgage file formats contain descriptive metadata that can be essential, the bulk of a picture file is just arrays of numeric infomation that when interpreted according to a certain rule set becomes recognizable as an image to us. Our image of an eight is no exception, and skimage.io stored that image data in an array of arrays making a 5 x 3 matrix of 15 pixels. We can demonstrate that by calling on the shape properity of our image variable and see the matrix by printing our image variable to the screen.
+
+
+ ~~~
+print(image.shape)
+print(image)
+~~~
+{: .language-python}
+![Image of 8 matrix](../fig/02-matrix-8.png)
+
+  Thus if we have tools that will allow us to manipulate these arrays of numbers, we can manipulate the image.  The numpy library can be particularly useful here, so let's try that out using numpy array slicing. Notice that the default behavior of the imageshow function appended row and an column numbers that will be helpful to us as we try to address indiviual or groups of pixels. First lets' load another copy of our eight, and then make it look like a zero.
+
+To make it look like a zero, we need to change the number underlying the center most pixel to be 1.  With the help of those row and column headers, at this small scale we can determine  the center pixel is in row labled 2 and column labeled 1. Using array slicing, we can then address and assign a new value to that position.
+
+ ~~~
+zero = skimage.io.imread(fname="data/eight.tif")
+zero[2,1]= 1.0
+"""
+The follwing line of code creates a new figure for imshow to use in displaying our output. Without it, plt.imshow() would overwrite our previous image in the cell above
+""" 
+fig, ax = plt.subplots()
+plt.imshow(zero)
+print(zero)
+~~~
+{: .language-python}
+
+![Image of 0 matrix](../fig/02-matrix-0.png)
+![Image of 0](../fig/02-zero.png)
```

Comment:
This image appears in black and white. Perhaps just a slip, but I'd suggest to stick to same colormap for the entire section.

Reply:
Agreed!

Reply:
implemented